cupy/testing/helper.py

```diff
@@ -590,6 +588,18 @@ def numpy_cupy_raises(name='xp', sp_name=None, scipy_name=None,
     assert sp_name is None or isinstance(sp_name, str)
     assert scipy_name is None or isinstance(scipy_name, str)
 
+    while True:
+        if (isinstance(accept_error, type)
+                and issubclass(accept_error, Exception)):
+            break
+        if (isinstance(accept_error, tuple)
+                and all(isinstance(ex, type) for ex in accept_error)
+                and all(issubclass(ex, Exception) for ex in accept_error)):
+            break
+        raise TypeError(
+            'accept_error must be a subtype of Exception or a tuple of such '
+            'types.')
```

Comment:
Why did you use a while-loop? This loop looks iterated at most once.
How about writing like as follows?
```py
if not isinstance(accept_error, tuple):
    accept_error = accept_error,
if not all(isnstance(ex, type) and issubclass(ex, Exception) for ex in accept_error):
    raise TypeError(...)
```

Reply:
IMO `whlie True:` is cleaner. It's more readable and there's no redundancy (no need to preprocess before checking).
I often write this way personally, but I will fix it if you still don't like it.

Reply:
There are two similar fragment `isinstance(...) and issubclass(...)` in your implementation, but it is avoidable by introducing preprocess before checking. I still feel my suggestion is better.

Reply:
OK. By the way, let's settle the discussion of #2871 before making a fix.